apps/settings/js/modules/bluetooth/bluetooth_context.js

```diff
@@ -0,0 +1,852 @@
+/* global console */
+
+/**
+ * BluetoothContext:
+ *   - BluetoothContext is an Observable that wraps the platform Bluetooth
+ *     object.
+ *   - BluetoothContext is a singleton that you can easily use it to fetch some
+ *     shared data across different panels.
+ *   - It has some observable properties: state, enabled, address, name,
+ *     discoverable, discovering, numberOfPairedDevices, firstPairedDeviceName,
+ *     hasPairedDevice.
+ *   - It has two observable array: _pairedDevices, _remoteDevices.
+ * BluetoothContext only update state and does not involve in any UI logic.
+ *
+ * @module BluetoothContext
+ */
+define(function(require) {
+  'use strict';
+
+  var AdapterManager = require('modules/bluetooth/bluetooth_adapter_manager');
+  var Observable = require('modules/mvvm/observable');
+  var ObservableArray = require('modules/mvvm/observable_array');
+  var SettingsCache = require('modules/settings_cache');
+
+  var settings = Settings.mozSettings;
```

Comment:
Please use `navigator.mozSettings` because in the long term we are going to remove `Settings` from window.